main/src/library/RedBlackTree.flix

```diff
@@ -178,17 +178,16 @@ namespace RedBlackTree {
         case _ => s
     }
 
-    ///
     /// Applies `f` to all key-value pairs in `tree` going from left to right until a single pair `(k, v)` is obtained.
     ///
     /// That is, the result is of the form: `Some(f(...f(f(k1, v1, k2, v2), k3, v3)..., kn, vn))`
     ///
     /// Returns `None` if `tree` is the empty tree.
     ///
     pub def reduceLeft(f: (k, v, k, v) -> (k, v) & e, tree: RedBlackTree[k, v]): Option[(k, v)] & e =
-        reduceLeftHelper((x: Option[(k, v)], y: (k, v)) -> match x {
-            case Some((k, v)) => Some(f(k, v, fst(y), snd(y)))
-            case None => Some(y)
+        foldLeft((x: Option[(k, v)], k1: k, v1: v) -> match x {
```

Comment:
Did you put the type annotation to help you? Or did it simply not compile without it?

Reply:
No idea, an old habit from Scala maybe.
I have tested it without and removed it.